update_jets.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+set -e
+
+
+if [ -z "$1" ]; then
+  echo "\$1 parameter must be the simplicity directory"
+  echo "\$2 parameter (optional) can be the revision to check out. If this is not specified, we
+  use the simplicity revision specified in the simplicity-sys/depend/simplicity-HEAD-revision.txt.
+  If this is specified the revision must be available in the simplicity directory."
+  exit 1
+fi
+
+C_DIR=$1
+REV=$2
+RUST_DIR=$(pwd)
+VENDORED_HEAD=$(sed -n '2p' "$RUST_DIR"/simplicity-sys/depend/simplicity-HEAD-revision.txt)
+
+cd "$C_DIR" || exit 1
+if [ -n "$REV" ]; then
+    if [ "$REV" != "$VENDORED_HEAD" ]; then
+        echo "WARNING: Simplicity GenRustJets and simplicity depend do not have the same git rev"
+        echo "$REV"
+        echo "$VENDORED_HEAD"
+    fi
+    git checkout "$REV"
+else
+    git checkout "$VENDORED_HEAD"
```

Comment:
In 28ee0c22fb42b5eb7c26371e67846aca4280be79:

I still don't like that we're running `git checkout` which is potentially destructive. But offhand I don't know what the "correct" solution is (probably something obscure like `git read-tree --prefix`) and anyway I'd like to get this PR in. So we can address it in a followup.